internetee/eis_billing_system
# added eeid auth

**What is this?**
To enhance security, I decided to implement our EEID authentication system. This system allows authorization based solely on personal data.

**What did you do?**
I integrated this system into the billing system. I also added a WhiteCode table, which allows adding a personal code to the table that is authorized to log in to the system. If a different personal code is specified during authorization, which is not present in WhiteCode, a 'forbidden' error appears.

I also added a special job that allows you to add a personal code to the database through the console; the syntax looks like this:
`rake whitelist:add_code[60001019906]`

Additionally, in the admin user table, the user can also add personal codes for those participants who will be allowed to log in using their personal documents.

**How to test this?**
Testing should be done carefully, as this system solves a security problem. It is essential to ensure that security has genuinely improved rather than the opposite. In addition to standard authorization with a code that is present and absent in the whitelist, it is also necessary to check the operation of the API. Make sure that access is denied without a special token, and that access to other pages without authorization is also prohibited.

**How to review the code?**

- Ensure that the tests cover all the necessary functionality, and that security tests are also present.
- Make sure the code is clear and clean. The code should not be confusing and should appear straightforward and simple, making it easy to understand what it does.
- Ensure that the code will not cause any other potential errors.
